### ISSUE TITLE
Allow for generic schedulers in automatic mode

### ISSUE THREAD

**Issue by thoglu** [PyTorchLightning/pytorch-lightning#9335]
In  the configure_optimizers function (https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/optimizers.py#L123) a check is made if automatic or manual optimization is used. For manual mode, you can use any scheduler object, as in https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/optimizers.py#L138 the generic scheduler object is loaded along with its config.

In automatic mode, generic schedulers are currently not supported: they must be of class ReduceLROnPlateau, or subclass _LRScheduler. 

Why must a custom scheduler imply that you use manual optimization mode? 
I think we should also allow custom schedulers in automatic mode.

**Comment by tchaton** [comment]
Dear @thoglu,

Mind sharing more details on what you want to achieve ? We force `_LRScheduler`, so we can ensure the right behaviour for the schedulers + typing.

Best,
T.C

**Comment by thoglu** [comment]
Hi,
I have a custom scheduler that does not inherit from _LRScheduler but would like to use it in automatic mode. I now changed it to actually inherit from _LRScheduler and it is fine, but I had to hack the "step" method because it is called ẃithin __init__ of _LRScheduler and that for me is unwanted behavior. So I have a solution for me that works now, and have no strong opinion on this issue.
Best,
thoglu
